python/setup.py

```diff
@@ -41,14 +41,15 @@
     packages=[
         'delta_sharing',
     ],
-    python_requires='>=3.7',
+    python_requires='>=3.8',
     install_requires=[
+        'delta_kernel_rust_sharing_wrapper',
         'pandas',
-        'pyarrow>=4.0.0',
+        'pyarrow>=16.1.0',
         'fsspec>=0.7.4',
         'requests',
         'aiohttp',
-        'dataclasses;python_version<"3.7"',
+        'dataclasses;python_version<"3.8"',
         'yarl>=1.6.0',
         'maturin',
```

Comment:
we might be able to remove maturin here since it is only used to build delta_lernel_rust_sharing_wrapper but this is now being done in python_release.sh. Not totally sure though but if we can avoid having user install an extra package this might be a good thing

Reply:
updated